### PR TITLE
Feat: gzip/decompressed response content handling 

### PR DESCRIPTION
**Pull request by anibadde**

### CODE REVIEW

File: google/oauth2/id_token_async.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
+# Copyright 2020 Google LLC
```
Comment (crwilcox) [inline google/oauth2/id_token_async.py:1]: @anibadde what is this addition from?

File: google/auth/transport/aiohttp_requests.py [hunk 2]
```diff
@@ -31,6 +32,51 @@
 _DEFAULT_TIMEOUT = 180  # in seconds
 
 
+class _CombinedResponse(transport.Response):
```
Comment (crwilcox) [inline google/auth/transport/aiohttp_requests.py:35]: TODO: we are seeing if this can't be pushed into resumable

File: google/auth/transport/aiohttp_requests.py [hunk 3]
```diff
@@ -230,8 +280,14 @@ async def request(
                 transmitted. The timout error will be raised after such
                 request completes.
         """
-
-        async with aiohttp.ClientSession() as self._auth_request_session:
+        if headers:
+            for key in headers.keys():
```
Comment (crwilcox) [inline google/auth/transport/aiohttp_requests.py:284]: TODO: is this still an issue?

File: google/auth/transport/aiohttp_requests.py (hunk 2 above)
Comment (crwilcox) [reply google/auth/transport/aiohttp_requests.py:35]: made issue to track this.

File: google/auth/transport/aiohttp_requests.py [hunk 4]
```diff
@@ -31,6 +32,51 @@
 _DEFAULT_TIMEOUT = 180  # in seconds
 
 
+class _CombinedResponse(transport.Response):
+    '''
+    In order to more closely resemble the `requests` interface, where a raw
+    and deflated content could be accessed at once, this class lazily reads the 
+    stream in `transport.Response` so both return forms can be used
+    '''
```
Comment (tritone) [inline google/auth/transport/aiohttp_requests.py:40]: Use double quotes for docstring. Does this need to be fleshed out more?

File: google/auth/transport/aiohttp_requests.py [hunk 5]
```diff
@@ -31,6 +32,51 @@
...
+    def __init__(self, response):
+        self._response = response
+        self._raw_content = None
+
+    def _is_compressed(self):
+        # The gzip and deflate transfer-encodings are automatically decoded for you.
```
Comment (tritone) [inline google/auth/transport/aiohttp_requests.py:46]: This comment is a bit unclear, maybe rephrase.

File: google/auth/transport/aiohttp_requests.py [hunk 6]
```diff
@@ -230,8 +280,16 @@ async def request(
                 transmitted. The timout error will be raised after such
                 request completes.
         """
+        # Headers come in as bytes which isn't expected behavior (resumable)
```
Comment (tritone) [inline google/auth/transport/aiohttp_requests.py:283]: What do you mean by "isn't expected behavior"? Can you clarify to explain why the lines below are necessary?

File: google/auth/transport/aiohttp_requests.py [hunk 7]
```diff
@@ -264,7 +322,7 @@ async def request(
                     data=data,
                     headers=request_headers,
                     timeout=timeout,
-                    **kwargs
+                    **kwargs,
```
Comment (tritone) [inline google/auth/transport/aiohttp_requests.py:325]: Get rid of stray comma here and below. In general, it's good to review your diff to make sure you're only including changes that you really want to include.

File: google/auth/transport/aiohttp_requests.py (hunk 6 above)
Comment (crwilcox) [reply google/auth/transport/aiohttp_requests.py:283]: This isn't expected as most headers are of a str type. It is odd that some come through as bytes. This may be a bandaid, or may be entirely necessary :(

File: google/auth/transport/aiohttp_requests.py (hunk 7 above)
Comment (crwilcox) [reply google/auth/transport/aiohttp_requests.py:325]: I think this may be the autoformatter at work.  Might be worth running `nox -s blacken` across code quick actually, might be a differing version?

Comment (crwilcox) [reply google/auth/transport/aiohttp_requests.py:325]: confirmed this is autoformatter (black)

File: google/oauth2/_client_async.py [hunk 8]
```diff
@@ -104,6 +104,7 @@ async def _token_endpoint_request(request, token_uri, body):
             method="POST", url=token_uri, headers=headers, body=body
         )
 
+        # Using data.read() causing zlib decompression error
```
Comment (crwilcox) [inline google/oauth2/_client_async.py:107]:
```suggestion
        # Using data.read() resulted in zlib decompression errors. This may require future investigation.
```

File: google/auth/transport/aiohttp_requests.py (hunk 5 above)
Comment (crwilcox) [reply google/auth/transport/aiohttp_requests.py:47]: possibly move this into the comment up top as this is the point of this added class.

File: google/auth/transport/aiohttp_requests.py (hunk 4 above)
Comment (anibadde) [reply google/auth/transport/aiohttp_requests.py]: Resolved, changed to double quotes with blacken.

File: google/auth/transport/aiohttp_requests.py (hunk 5 above)
Comment (anibadde) [reply google/auth/transport/aiohttp_requests.py:47]: Fleshed out this comment in the latest version to explain reasoning.

File: google/auth/transport/aiohttp_requests.py (hunk 6 above)
Comment (anibadde) [reply google/auth/transport/aiohttp_requests.py]: Added more explanation to why we are doing this. Inconsistencies in the header value types for some cases in the resumable library, which is why we have this check.
